src/main/java/net/neoforged/neoforge/common/extensions/IItemExtension.java

```diff
@@ -457,13 +458,14 @@ default int getEnchantmentLevel(ItemStack stack, Holder<Enchantment> enchantment
      * Used in several places in code including armor enchantment hooks.
      * The returned value(s) must have the same level as {@link #getEnchantmentLevel(ItemStack, Enchantment)}.
      *
-     * @param stack The item stack being checked
+     * @param stack  The item stack being checked
+     * @param lookup A registry lookup, used to resolve enchantment {@link Holder}s.
      * @return Map of all enchantments on the stack, empty if no enchantments are present
      * @see #getEnchantmentLevel(ItemStack, Enchantment)
      * @apiNote Call via {@link IItemStackExtension#getAllEnchantments()}.
      */
     @ApiStatus.OverrideOnly
-    default ItemEnchantments getAllEnchantments(ItemStack stack) {
+    default ItemEnchantments getAllEnchantments(ItemStack stack, RegistryLookup<Enchantment> lookup) {
         return stack.getOrDefault(DataComponents.ENCHANTMENTS, ItemEnchantments.EMPTY);
```

Comment:
I think it might be slightly clearer what this is doing if we have the default implementation call `ItemStack#getTagEnchantments`